### PR TITLE
linux package

### PR DESCRIPTION
**Pull request by altruios**
For installing using: 
```
makepkg -si
```
On linux systems (that have makepkg).

That requires one .sh file to call the java -jar particle-life.jar command.
(Well, 'requires' is a strong word)
This is not a finished draft - but it's a working start (**requires a release of 1.1.3** with this new code to work - I can't test 1.1.2 directly as that is before Linux support), there's a bit more info and boilerplate to add to make it compliant with AUR standards..

Tested using my fork and it all seems to be well: I'm sure that there's things to fix.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,6 +5,8 @@ plugins {
     id "application"                            // for "run" task
     id 'edu.sc.seis.launch4j' version '2.5.1'   // for generating .exe
     id 'org.beryx.runtime' version '1.12.7'     // for generating JRE
+   id 'com.github.johnrengelman.shadow' version '7.1.2' //to make linux build
```
Comment (tom-mohr) [inline build.gradle:8]: change comment `// only needed to make linux build`

File: PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Maintainer: Your Name <youremail@domain.com>
+pkgname="particle-life"
+pkgrel=1
+pkgver=1.1.3 # todo: make automatic
+pkgdesc="GUI for Particle Life, a particle system showing life-like behaviour"
+arch=("x86_64")
+url="https://github.com/tom-mohr/particle-life-app"
+license=('GPL')
+depends=('java-environment')
+makedepends=('git' 'java-environment>=11')
```
Comment (tom-mohr) [inline PKGBUILD:10]: The Particle Life App requires Java >= 16, so this here should say `java-environment>=16`, I assume?

File: PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Maintainer: Your Name <youremail@domain.com>
+pkgname="particle-life"
+pkgrel=1
+pkgver=1.1.3 # todo: make automatic
```
Comment (tom-mohr) [inline PKGBUILD:4]: The latest version of Particle Life App is currently 1.1.2, but I assume that you wait for 1.1.3 which includes the changes that make it work on linux, right?

File: PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+pkgdesc="GUI for Particle Life, a particle system showing life-like behaviour"
+arch=("x86_64")
+url="https://github.com/tom-mohr/particle-life-app"
+license=('GPL')
+depends=('java-environment')
+makedepends=('git' 'java-environment>=11')
+source=("$pkgname.tar.gz::$url/archive/refs/tags/$pkgver.tar.gz")
```
Comment (tom-mohr) [inline PKGBUILD:11]: I think there is a "v" missing before $pkgver

File: PKGBUILD (hunk 2 above)
Comment (altruios) [reply PKGBUILD:10]: yep: that it should be

File: PKGBUILD (hunk 3 above)
Comment (altruios) [reply PKGBUILD:4]: aye - there is a way to make that into a fn call to always be at the latest release: I have not gotten that to actually work yet.

File: PKGBUILD (hunk 4 above)
Comment (altruios) [reply PKGBUILD:11]: aye - my test releases did not. oversight on my part :)

File: PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Maintainer: Your Name <youremail@domain.com>
+pkgname="particle-life"
+pkgrel=1
+pkgver=v1.1.3 # todo: make automatic
```
Comment (tom-mohr) [inline PKGBUILD:4]: According to [this](url), you can automate this by defining a `pkgver()` function in the PKGBUILD
Example from https://wiki.archlinux.org/title/VCS_package_guidelines#The_pkgver()_function:
```sh
pkgver() {
  cd "$pkgname"
  git describe --long | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
}
```

**Comment by altruios** [comment]
I am closing this branch to make a better PR - just for the PKGBUILD file

**Pull request closed without merging**